lib/handlers/quote/quote.ts

```diff
@@ -245,6 +245,12 @@ export class QuoteHandler extends APIGLambdaHandler<
       ...(enableFeeOnTransferFeeFetching ? { enableFeeOnTransferFeeFetching } : {}),
     }
 
+    if (enableFeeOnTransferFeeFetching) {
+      routingConfig.writeToCachedRoutes = true
```

Comment:
are we sure we want to enable `writeToCachedRoutes` just because we have `enableFeeOnTransferFeeFetching`? Shouldn't this be determined by the cached routes logic itself?

Reply:
Actually we don't need to hardcode `writeToCachedRoutes` here, because sor hardcoded the final `AlphaRouterConfig`:

```
const routingConfig: AlphaRouterConfig = _.merge(
      {
        // These settings could be changed by the partialRoutingConfig
        useCachedRoutes: true,
        writeToCachedRoutes: true,
        optimisticCachedRoutes: false,
      },
      DEFAULT_ROUTING_CONFIG_BY_CHAIN(this.chainId),
      partialRoutingConfig,
      { blockNumber }
    );
```

I think we need to hardcode `routingConfig.useCachedRoutes = false` only

Reply:
alright shall we fix it then?

Reply:
Yes, I actually fixed it in a better way I think:

```
export const FEE_ON_TRANSFER_SPECIFIC_CONFIG = (enableFeeOnTransferFeeFetching?: boolean) => {
  return {
    // if enableFeeOnTransferFeeFetching is true, then we do not use cached routes for read path
    // if enableFeeOnTransferFeeFetching is false or undefined, then we use cached routes for read path
    useCachedRoutes: !enableFeeOnTransferFeeFetching,
    enableFeeOnTransferFeeFetching: enableFeeOnTransferFeeFetching,
  }
}
```

Then in quote.ts:

```
const routingConfig: AlphaRouterConfig = {
      ...DEFAULT_ROUTING_CONFIG_BY_CHAIN(chainId),
      ...(minSplits ? { minSplits } : {}),
      ...(forceCrossProtocol ? { forceCrossProtocol } : {}),
      ...(forceMixedRoutes ? { forceMixedRoutes } : {}),
      protocols,
      ...(quoteSpeed ? QUOTE_SPEED_CONFIG[quoteSpeed] : {}),
      ...parsedDebugRoutingConfig,
      // enable fee-on-transfer needs to be before intent specific config,
      // in case someone wants to override useCachedRoutes with intent other than caching
      // we allow reading from the cached routes
      ...FEE_ON_TRANSFER_SPECIFIC_CONFIG(enableFeeOnTransferFeeFetching),
      ...(intent ? INTENT_SPECIFIC_CONFIG[intent] : {}),
    }
```

This way if someone wants to issue `intent` along with `enableFeeOnTransferFeeFetching`, the `useCachedRoutes` should be true, since `intent` takes precedence over `enableFeeOnTransferFeeFetching`